File: libgomp/testsuite/libgomp.oacc-fortran/lib-5.f90
! { dg-do run }

program main
  use openacc
  implicit none

  integer n

  if (acc_get_num_devices (acc_device_nvidia) .ne. 0) then

     call acc_init (acc_device_nvidia)

     n = 0

     call acc_set_device_num (n, acc_device_nvidia)

     if (acc_get_device_num (acc_device_nvidia) .ne. 0) stop 11

     if (acc_get_num_devices (acc_device_nvidia) .gt. 1) then

        n = 1

        call acc_set_device_num (n, acc_device_nvidia)

        if (acc_get_device_num (acc_device_nvidia) .ne. 1) stop 12

     end if

     call acc_shutdown (acc_device_nvidia)

  end if

  if (acc_get_num_devices (acc_device_radeon) .ne. 0) then

     call acc_init (acc_device_radeon)

     n = 0

     call acc_set_device_num (n, acc_device_radeon)

     if (acc_get_device_num (acc_device_radeon) .ne. 0) stop 21

     if (acc_get_num_devices (acc_device_radeon) .gt. 1) then

        n = 1

        call acc_set_device_num (n, acc_device_radeon)

        if (acc_get_device_num (acc_device_radeon) .ne. 1) stop 22

     end if

     call acc_shutdown (acc_device_radeon)

  end if

end program

File: libgomp/testsuite/libgomp.oacc-c-c++-common/lib-5.c
/* { dg-do run } */

#include <stdlib.h>
#include <openacc.h>

int
main (int argc, char **argv)
{
  if (acc_get_device_type () == acc_device_default)
    abort ();

  acc_init (acc_device_default);

  if (acc_get_device_type () == acc_device_default)
    abort ();

  acc_shutdown (acc_device_default);

  if (acc_get_num_devices (acc_device_nvidia) != 0)
    {
      acc_init (acc_device_nvidia);

      if (acc_get_device_type () != acc_device_nvidia)
        abort ();

      acc_shutdown (acc_device_nvidia);

      acc_init (acc_device_default);

      acc_set_device_type (acc_device_nvidia);

      if (acc_get_device_type () != acc_device_nvidia)
        abort ();

      acc_shutdown (acc_device_nvidia);
    }

  if (acc_get_num_devices (acc_device_radeon) != 0)
    {
      acc_init (acc_device_radeon);

      if (acc_get_device_type () != acc_device_radeon)
        abort ();

      acc_shutdown (acc_device_radeon);

      acc_init (acc_device_default);

      acc_set_device_type (acc_device_radeon);

      if (acc_get_device_type () != acc_device_radeon)
        abort ();

      acc_shutdown (acc_device_radeon);
    }

  return 0;
}
